src/base-command.ts

```diff
@@ -16,6 +20,15 @@ export default abstract class extends Command {
     help: flags.help({ char: 'h' }),
   };
 
+  checkFlagDeprecations(flags: any, flagDefinitions: any) {
+    Object.keys(flags).forEach((flagName: string) => {
+      const description: string = flagDefinitions[flagName].description
+      if (description?.startsWith(DEPRECATED_LABEL)) {
+        this.warn(`Flag --${flagName} is deprecated.${description.split(DEPRECATED_LABEL)[1]}`);
+      }
+    });
+  }
+
```

Comment:
I see my pseudo code didn't work first time. Use the following so you don't have to call checkFlagDeprecations in every command.

```ts
async init() {
    const { flags } = this.parse(this.constructor as any);
    const flag_definitions = (this.constructor as any).flags;
    this.checkFlagDeprecations(flags, flag_definitions);
    ...
```

Reply:
doah, that helps a bunch, thanks! Will get that switched over